src/lib/storages/RedisProxyCache.ts

```diff
@@ -181,11 +246,60 @@ export class RedisProxyCache implements IProxyCache {
     }
   }
 
+  private async processNode(node: Redis, options: ProcessNodeOptions): Promise<void> {
+    const { pattern: match, batchSize: count, callbackFn, resolve, reject } = options
+    const stream = node.scanStream({ match, count })
+
+    stream.on('data', async (keys) => {
+      stream.pause()
+      try {
+        await Promise.all(keys.map((key: string) => this.processKey(key, callbackFn)))
+      } catch (err: unknown) {
+        stream.destroy(err as Error)
+        reject(err)
+      }
+      stream.resume()
+    })
+    stream.on('end', resolve)
+  }
+
+  private async processKey(key: string, callbackFn: ProcessKeyCallback): Promise<any> {
```

Comment:
a couple of question about this function:

1. Am I right that we'll process here both "types" of keys:
 `als:${sourceId}:${type}:${partyId}` and `als:${sourceId}:${type}:${partyId}:expiredAt` ?
2. 
```
    const actualKey = key.replace(':expiresAt', '')
    const expiresAt = await this.redisClient.get(key)
```
  it seems for me that `expiresAt` will contain `uniqueProxyIds`, no?

Reply:
I'm sure that is obvious in the code. What concerns do you have?

Reply:
@oderayi why do we need to process `als:${sourceId}:${type}:${partyId}`  key ?

why not just change "expiration" key structure a bit to avoid processing "usual" key?

Reply:
@geka-evk What do you mean by 

> why not just change "expiration" key structure a bit to avoid processing "usual" key?

`processKey` accepts expiry key, deduces the structure of the 'actual` key from the expiry key by removing the `expiresAt` suffix. Reads the expiry key from Redis to check if it's expired. Delete both expiry and actual key if true.

What changes are you suggesting?